Keystone; Automatic deletion of containers.

cbartz opened the pull request:
- Support for keystone-authentication
- Automatic deletion of shared_containers, when the original container gets deleted or the acls are removed

Commit:
Keystone; Automatic deletion of containers.

Commit:
Keystone; Automatic deletion of containers.

Commit:
Keystone-client package optional.

Commit:
Keystone-client package optional.

cbartz commented:
Hello Christian,

the containers are all empty (0 objects), because they are just linking to the original container via a metadata entry. Thus, the deletion should not fail.

cschwede commented:
Yes, that's true in most cases. However, there are a few cornercases:
- a user has no permission to delete the alias container
- the alias container existed before setting the alias and contains at least one object
- the alias is removed and another user puts an object into it

In that case the middleware crashes and might crash the proxy as well, so it's better to be on the safe side.

cbartz commented:
Are you sure that the middleware crashes, when a subrequest (using make_pre_authed_request) fails? Isn't it more just the case, that the response code indicates a failure?

Commit:
Log error if keystone authentication requested but module not found

Commit:
Fix flake8 warnings, add tox.ini

Commit:
Check target container ACL

Some middlewares are not checking ACLs, for example formpost.

This patch compares the ACL of the target container and denies access if the
requesting user is not allowed to access the target container.

Commit:
Add support for tempurl & formpost

The user is unknown if a request is authenticated using tempurl or formpost. In
this case we know that the request is authenticated for the original account,
thus we can check if an ACL for that account matches and grant access in this
case.

Commit:
Strip reseller_prefix from account name when verifying tempurl ACL

Commit:
Fix formpost support

Commit:
Merge branch 'check-acl' of https://github.com/cschwede/swift-containeralias

Commit:
Fix bug trying to create/delete containers for non-existing account.

Commit:
Fix user_group bug for tempurl/formpost, multiple auth methods same time

Commit:
user_groups for keystone
